2020/cmd/08/vm.go

```diff
@@ -63,6 +64,9 @@ func parseInstruction(s string) (instr Instruction, err error) {
 
 	switch matches[1] {
 	case "nop":
+		// QUESTION: this one is weird to me. It does not compile if I expect `*Instruction` in params.
+		// Why?
```

Comment:
Because type `*nopInstr` (type "pointer to nopInstr") is not `*Instruction`. It is not pointer to interface.
However it implements `Instruction` interface, so it could be passed as "wrapped" in such interface. Why it implements this interface? It implements all its methods.

Moreover, `nopInstr` (type "nopInstr") does not implement `Instruction` interface. Why? It has no Exec method with value receiver.